clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/InternalLocalBootstrappingVeniceChangelogConsumer.java

```diff
@@ -121,6 +126,31 @@ private Function<String, Boolean> functionToCheckWhetherStorageEngineShouldBeKep
       // is by reading the local storagemetadata bootstrap coordinate, and see if the internal client is able to
       // subscribe to that position. If it's not able to, that means that the local state is off Venice retention,
       // and therefore should be completely rebootstrapped.
+      for (int partition = 0; partition < partitionCount; partition++) {
+        OffsetRecord offsetRecord = storageMetadataService.getLastOffset(LOCAL_STATE_TOPIC_NAME, partition);
+        VeniceChangeCoordinate localCheckpoint;
+        try {
+          localCheckpoint = VeniceChangeCoordinate.decodeStringAndConvertToVeniceChangeCoordinate(
+              offsetRecord.getDatabaseInfo().get(CHANGE_CAPTURE_COORDINATE));
+        } catch (IOException | ClassNotFoundException e) {
```

Comment:
Maybe be defensive to NPE, or for offsetREcord being null?

Reply:
Added a check of offsetRecord == null above.